src/protocol/fluvio-protocol-derive/src/de.rs

```diff
@@ -17,6 +17,10 @@ pub(crate) fn generate_decode_trait_impls(input: &DeriveItem) -> TokenStream {
             let (impl_generics, ty_generics, where_clause) = kf_struct.generics.split_for_impl();
             quote! {
                 impl #impl_generics fluvio_protocol::Decoder for #ident #ty_generics #where_clause {
+                    #[tracing::instrument(
+                        skip(self, src, version),
+                        fields(type_name = std::any::type_name::<#ident #ty_generics>()),
+                    )]
```

Comment:
this is too low level to add instrument?

Reply:
Part of the benefit of putting this here is that, since we manually control the Encoding/Decoding, we may want visibility into the performance of the encode/decode implementations for various pieces of data in case something is accidentally taking a long time when it shouldn't. That was my rationale at least. If you think it is too low-level I can remove it

Reply:
encode/decode is at `ns` resolution (this will be be compiler optimized) which is too low level which was not designed for tracing.